controller/project/manager.py

```diff
@@ -118,3 +123,68 @@ def get_confusion_matrix(
     return project.get_confusion_matrix(
         project_id, labeling_task_id, for_classification, slice_id
     )
+
+
+def resolve_request_huddle_data(
+    project_id: str, user_id: str, data_id: str, huddle_type: str
+) -> HuddleData:
+    huddle = HuddleData(huddle_type=huddle_type, start_pos=-1, can_edit=True)
+    if huddle_type == enums.LinkTypes.SESSION.value:
+        session = search.resolve_labeling_session(project_id, user_id, data_id)
+        huddle.record_ids = session.session_record_ids
+        if __no_huddle_id(data_id):
+            data_id = session.id
+    else:
+        source_type = enums.LabelSource.MANUAL
+        source_id = None
+        if __no_huddle_id(data_id):
+            data_id = __get_first_data_id(project_id, user_id, huddle_type)
+        if huddle_type == enums.LinkTypes.DATA_SLICE.value:
+            slice_id = data_id
+        elif huddle_type == enums.LinkTypes.HEURISTIC.value:
+            is_data = __get_crowd_label_is_data(project_id, data_id)
```

Comment:
we need to get away from information source abbreviation `is`. I was totally confused by `is_data` as a non-boolean value

Reply:
changed to information_source_data & information_source_manager